pkg/aws/gw-machineset.go

```diff
@@ -65,12 +65,14 @@ spec:
                 - name: tag:Name
                   values:
                     - {{.InfraID}}-worker-sg
+          {{if not .UseLoadBalancer}}
```

Comment:
```suggestion
          {{- if not .UseLoadBalancer }}
```

You need a `-` otherwise this will be rendered as an empty line, also space is necessary in this case.
https://pkg.go.dev/text/template#hdr-Text_and_spaces

You can test the template here - https://gotemplate.io/